### ISSUE TITLE
RetroArch's memory card behavior (A1 only or All Slot 1) for Dreamcast breaks behavior for Sonic Shuffle

### ISSUE THREAD

**Issue by GHNeko** [libretro/RetroArch#14362]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

Sonic Shuffle is a party game for the Sega Dreamcast. It's a first party game that makes heavy use of the VMU, but Sega realized that not everyone would have their LCD VMUs and coded behaviors that trigger based on what device is in slot 1 and if there is a device in slot 1.

Generally in VS mode, when you load into the game with a VMU in Slot 1, when the player is selecting cards, their hand is shown on their VMU screen instead of in game. However if a normal memory card is in slot 1 (no screen) or there is no device in slot 1 (VMU in slot 2) the game will instead display the player's hand on the screen.

This is important because while this is a non issue for single player play (as you can emulate the vmu display), for multiplayer purposes, emulating VMU display essentially is showing all players hands to everyone and that's not how you play this primarily multiplayer game lol.

If you do not emulate the VMU display, but the game still sees a VMU instead of an MU in slot 1, it will still hide the cards on screen and then as a result you cannot see what card you have, breaking a lot of the functionality of the game for multiplayer as the game's design is focused around knowing what you have and it is designed to be a game that's played with friends.

So tldr - Because RetroArch only emulates a VMU and not also an MU, and always has a VMU in Slot 1 of Controller A or Controllers A-D; you cannot trigger the contingency for when the game/console sees that there is either an incompatible device or no device in slot 1.

To further supplement this idea, if you play Story Mode (single player only experience), the players cards are always face up regardless of what's in Slot 1 as by design its a single player experience and there is no issue in having your cards visible.

### Expected behavior

This is not a bug with the game per say, but I expected RetroArch to either emulate a no-screen memory card or allow me to set no cards/devices to Slot 1 and instead read only from Slot 2.

### Actual behavior

As explained above, multiplayer is effectively crippled into an unplayable state as you can currently decide between everyone's hand being visible or no one's hand being visible, when the intended design of the game allows for card visibility on screen.

VMU Screen Display is not sufficient as you cannot make it so that only the appropriate player sees only their respective VMU screen, and similarly there is no support for using a real VMU on a real controller (like you can with the DreamConn)

### Steps to reproduce the bug

Play Sonic Shuffle VS mode with a memory card setup.

### Bisect Results

It's always been like this as far as I can tell as this is baked into RetroArch's design and handling of the memory cards.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: Any
- 
### Environment information

- OS: Windows 8.1 / Arch Linux

**Comment by bslenul** [comment]
This issues board is for RetroArch itself, not for the cores, you'll have better luck opening the issue on the Flycast repo: https://github.com/flyinghead/flycast/issues ;)

**Comment by GHNeko** [comment]
Flycast as a stand alone program doesn't have this issue because it let's you set memory cards to which ever slot. I was told about this by a member the last time I brought it up actually.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by bslenul** [comment]
Yes but both the standalone and the core are maintained from the repo I linked, so even if it's an issue that only affects RetroArch it should be posted there.

**Comment by GHNeko** [comment]
Understood! Thank you for the clarification. I had asked where to post this issue last night and was wasn't fully guided to the right location. I apologize. I was just told to post an issue to the github but not where specifically.

Sent via the Samsung Galaxy Z Fold3 5G, an AT&T 5G smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by bslenul** [comment]
No problem! Can you close this issue now, please?

**Comment by GHNeko** [comment]
Will do!
